src/main/java/ch/fhnw/algd1/converters/utf8/UTF8Converter.java

```diff
@@ -4,33 +4,70 @@
  * Created on 05.09.2014
  */
 /**
- * @author 
+ * @author
  */
 public class UTF8Converter {
 	public static byte[] codePointToUTF(int x) {
-		byte[] b = null;
-		// TODO allocate b in the right size (depending on x) and fill it with the
-		// UTF-8 encoding of code point x. b[0] shall contain the first byte.
-		return b;
+		if (x < 0 || x > 0b1111_1111_1111_1111_1111)
+			return null;
+
+		// 7 Bit -> 1 Byte, 11 Bit -> 2 Byte, 16 Bit -> 3 Byte, 21 Bit -> 4 Byte
+		int size = (x >>> 7 == 0) ? 1 : (x >>> 11 == 0) ? 2 : (x >>> 16 == 0) ? 3 : 4;
```

Comment:
Kann man so machen. Andererseits braucht es dann unten nochmal eine Fallunterscheidung und man kann sich fragen, ob man nicht die Unterscheidungskriterien von hier unten verwenden und dafür in den einzelnen Fällen die Instanziierung von b mit jeweils einer konstanten Grösse programmieren möchte. (Ist letztlich natürlich nicht "wichtig", käme mir aber irgendwie klarer vor.)